ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -142,6 +145,7 @@ public int validate(
     } else if (KsqlEngine.isExecutableStatement(configured.getStatement())
         || configured.getStatement() instanceof TerminateQuery) {
       final ConfiguredStatement<?> statementInjected = injector.inject(configured);
+      validateSupportedKeyFormat(statementInjected);
```

Comment:
Is the RequestValidator the right place to perform this validation? This felt more correct to me than either the executor or the DefaultFormatInjector, but given that neither QTT nor YATT calls the RequestValidator, I'm no longer sure.